samples/java_springboot/14.nlp-with-dispatch/src/main/java/com/microsoft/bot/sample/nlpwithdispatch/DispatchBot.java

```diff
@@ -28,25 +25,24 @@
 
 public class DispatchBot extends ActivityHandler {
 
-    private final Logger _logger;
-    private final BotServices _botServices;
+    private final Logger logger;
+    private final BotServices botServices;
 
     public DispatchBot(BotServices botServices) {
-        _logger = LoggerFactory.getLogger(DispatchBot.class);
-        _botServices = botServices;
+        logger = LoggerFactory.getLogger(DispatchBot.class);
+        this.botServices = botServices;
     }
 
     @Override
     protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
-        // First, we use the dispatch model to determine which cognitive service (LUS or
-        // QnA) to use.
-        RecognizerResult recognizerResult = _botServices.getDispatch().recognize(turnContext).join();
+        // First, we use the dispatch model to determine which cognitive service (LUIS or QnA) to use.
+        return botServices.getDispatch().recognize(turnContext).thenCompose(result -> {
```

Comment:
```suggestion
        return botServices.getDispatch().recognize(turnContext).thenCompose(recognizerResult -> {
```

Before was `recognizerResult`